language/libra-vm/src/libra_vm.rs

```diff
@@ -238,18 +238,14 @@ impl LibraVM {
     fn check_change_set(&self, change_set: &ChangeSet, state_view: &dyn StateView) -> VMResult<()> {
         // This function is only invoked by WaypointWriteSet for now. We don't enforce the same
         // check on TransactionPayload::WriteSet.
-        if state_view.is_genesis() {
```

Comment:
Is this change related?

Reply:
well, this is for the db-bootstrapper to be able to run a writeset txn on an existing state. We'll need to update the Executor accordingly so that it's still impossible to do it with normal consensus based flow.

Also, I'm not sure if it's entirely reasonable to "ban" deletions in such transactions? (I mean we are unleashing the devil through this txn anyways.)

Reply:
Good point; the "no deletions" check only makes sense for genesis.

Reply:
Do we want to just get rid of this check entirely given that we need to use this mechanism other than genesis?

Reply:
Yes, I think so. But I'm still wondering if removing the `is_genesis` check is related to the stated purpose of the PR?

Reply:
Yeah removing the check and adding the read are both my requests, but can be separate commits.

Reply:
No need to split; just wanted to make sure I understood the relationship--thanks!